addons/mail/static/src/discuss/call/public_web/discuss_sidebar_call_participants.js

```diff
@@ -1,23 +1,68 @@
 import { Component, useState } from "@odoo/owl";
 import { useService } from "@web/core/utils/hooks";
 import { Thread } from "@mail/core/common/thread_model";
+import { callActionsRegistry } from "../common/call_actions";
+import { useHover } from "@mail/utils/common/hooks";
+import { useDropdownState } from "@web/core/dropdown/dropdown_hooks";
+import { Dropdown } from "@web/core/dropdown/dropdown";
 /**
  * @typedef {Object} Props
  * @property {import(models").Thread} thread
  * @extends {Component<Props, Env>}
  */
 export class DiscussSidebarCallParticipants extends Component {
     static template = "mail.DiscussSidebarCallParticipants";
-    static props = { thread: { type: Thread } };
-    static components = {};
+    static props = { thread: { type: Thread }, compact: { type: Boolean, optional: true } };
+    static components = { DiscussSidebarCallParticipants, Dropdown };
 
     setup() {
         super.setup();
+        this.store = useState(useService("mail.store"));
         this.rtc = useState(useService("discuss.rtc"));
+        this.hover = useHover(["root", "floating*"], () => {
+            this.floating.isOpen = this.hover.isHover;
+        });
+        this.floating = useDropdownState();
+    }
+
+    get callActionsRegistry() {
+        return callActionsRegistry;
+    }
+
+    get compact() {
+        if (typeof this.props.compact === "boolean") {
+            return this.props.compact;
+        }
+        return this.store.discuss.isSidebarCompact;
+    }
+
+    get lastActiveSession() {
+        /** @type {import("models").Thread} */
+        const thread = this.props.thread;
+        const sessions = [...thread.rtcSessions];
+        sessions?.sort((s1, s2) => {
+            if (s1.isActuallyTalking && !s2.isActuallyTalking) {
+                return -1;
+            }
+            if (!s1.isActuallyTalking && s2.isActuallyTalking) {
+                return 1;
+            }
+            if (s1.isVideoStreaming && !s2.isVideoStreaming) {
+                return -1;
+            }
+            if (!s1.isVideoStreaming && s2.isVideoStreaming) {
+                return 1;
+            }
+            return s2.talkingTime - s1.talkingTime;
+        });
+        return sessions[0];
     }
 
-    get participants() {
-        return this.props.thread.rtcSessions?.sort((s1, s2) => {
+    get sessions() {
+        /** @type {import("models").Thread} */
+        const thread = this.props.thread;
+        const sessions = [...thread.rtcSessions];
```

Comment:
```suggestion
        const sessions = [...this.props.thread.rtcSessions];
```

Reply:
I did this on purpose for proper typing of `thread`. This helps with making less programming errors :)

Reply:
In this case, it would be better to directly type de props of the component

Reply:
This has been done in this component, but somehow it doesn't work. `LinkPreviewConfirmDelete` component did the same and it works. If you know why the typing of prop doesn't work here then no problem for the change :)

Reply:
Good catch, thanks!